### PR TITLE
feat: add extra maps

### PR DESCRIPTION
**Pull request by unclFedor**

### CODE REVIEW

**Review by belom88 — CHANGES_REQUESTED** [review]
A lot of excess code we should get rid of in this PR.

File: .env.local [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VITE_REACT_APP_MAPBOX_TOKEN=pk.eyJ1IjoiZHNhdmlub3YiLCJhIjoiY2xsd2VocjRsMjVpbzNwcDhsc3E1ZzAzNiJ9.3xSNq_gwuOMHPUTDkO4OzQ
+VITE_REACT_APP_REST_API_BASE_URL=https://developer.trimet.org/ws/v2
+VITE_REACT_APP_REST_API_APP_ID=97B28D2ED40DA08AB53687356
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply .env.local:3]: removed .env.local

File: .env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VITE_REACT_APP_MAPBOX_TOKEN=
+VITE_REACT_APP_REST_API_BASE_URL=https://developer.trimet.org/ws/v2
+VITE_REACT_APP_REST_API_APP_ID=
+VITE_SOME_KEY=hello
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply .env:4]: 1. removed VITE_ prefix in the .env content
2. uninstalled "vite" 
3. installed "dotenv". It seems "@types/dotenv" is deprecated
4. updated webpack.dev.config.js

File: package.json [hunk 3]
```diff
@@ -31,24 +38,29 @@
     "@loaders.gl/tiles": "^4.0.0-beta.2",
     "@luma.gl/core": "^8.5.14",
     "@math.gl/proj4": "^3.6.3",
+    "@mui/material": "^5.14.13",
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply package.json:41]: 1. removed @mui/material  from package.json
2. removed loading.tsx, unsupported.tsx
3. updated map-wrapper.tsx

File: package.json [hunk 4]
```diff
@@ -31,24 +38,29 @@
...
     "@probe.gl/stats": "^4.0.4",
     "@reduxjs/toolkit": "^1.9.5",
+    "@types/google.maps": "^3.54.4",
     "deck.gl": "^8.8.3",
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply package.json:45]: 1. removed "deck.gl": "^8.8.3"
2. added 
"@deck.gl/geo-layers": "^8.9.31",
"@deck.gl/layers": "^8.9.31",
"@deck.gl/mesh-layers": "^8.9.31",
"@deck.gl/react": "^8.9.31", 
"@loaders.gl/images": "^4.0.0-beta.2",
"@loaders.gl/schema": "^4.0.0-beta.2",
"apache-arrow": "^13.0.0",

File: src/components/interleaved-map/interleaved-map.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
+import { Map as MapboxMap } from "mapbox-gl";
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/components/interleaved-map/interleaved-map.tsx:1]: removed interleaved-map.tsx

File: src/components/loading/loading.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import CircularProgress from "@mui/material/CircularProgress";
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/components/loading/loading.tsx:1]: 1. Removed "@mui/material 
2. Removed loading .tsx

File: src/components/unsupported/unsupported.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import Paper from "@mui/material/Paper";
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/components/unsupported/unsupported.tsx:1]: removed unsupported.tsx

File: src/constants/notification-messages.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export const TERRAIN_WITH_SELECTED_BASE_MAP_PROVIDER_IS_NOT_SUPPORTED = (
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/constants/notification-messages.ts:1]: removed notification-messages.ts

File: src/components/deck-gl-wrapper/deck-gl-wrapper.tsx [hunk 9]
```diff
@@ -757,27 +828,40 @@ export const DeckGlWrapper = ({
...
+            maxzoom={12}
+          />
+        )}
+        {baseMapProviderId === BaseMapProviderId.mapbox2 && (
+          <MapboxSource
+            id={mapProviderProps.terrainProps.id}
+            type={mapProviderProps.terrainProps.type}
+            url="mapbox://mapbox.mapbox-terrain-dem-v1"
+            tileSize={512}
+            maxzoom={14}
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/components/deck-gl-wrapper/deck-gl-wrapper.tsx:861]: replaced to {mapProviderProps[BaseMapProviderId.mapbox2].terrainProps.XXX} and {mapProviderProps[BaseMapProviderId.maplibre].terrainProps.XXX} accordingly.

File: src/components/deck-gl-wrapper/deck-gl-wrapper.tsx [hunk 10]
```diff
@@ -212,8 +231,60 @@ export const DeckGlWrapper = ({
...
-  const showTerrain=selectedBaseMap?.id === "Terrain";
-  const mapStyle=selectedBaseMap?.mapUrl;
+  const showTerrain = selectedBaseMap?.id === "Terrain";
+  const mapStyle = selectedBaseMap?.mapUrl;
+  ///
+  const baseMapProvider = useAppSelector(selectMapProvider);
+  const baseMapProviderId =
+    baseMapProvider.id === BaseMapProviderId.maplibre
+      ? BaseMapProviderId.maplibre
+      : BaseMapProviderId.mapbox2;
```
[the remark this answers is not in the document]
Comment (unclFedor) [reply src/components/deck-gl-wrapper/deck-gl-wrapper.tsx:241]: removed expression. Changed type from String to BaseMapProviderId

**Pull request closed without merging**